src/main/scala/com/cognite/sdk/scala/v1/containers/ContainerPropertyType.scala

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 Cognite AS
+// SPDX-License-Identifier: Apache-2.0
+
+package com.cognite.sdk.scala.v1.containers
+
+import cats.implicits._
+import io.circe._
+import io.circe.generic.semiauto._
+
+sealed trait ContainerPropertyType
+
+object ContainerPropertyType {
+
+  final case class TextProperty(
+      list: Option[Boolean] = Some(false),
+      collation: Option[String] = Some("ucs_basic")
+  ) extends ContainerPropertyType {
+    val `type`: String = TextProperty.Type
```

Comment:
the `type` value is used by the api to identify the property. 
We don't need that value, but need to make sure that the user is not able to modify it in anyway.